### PR TITLE
[DI] Added _instanceof example

### LINKED ISSUE #7538: Document the new _instanceof section for services configuration
See https://github.com/symfony/symfony/pull/21530

### PR DESCRIPTION
**Pull request by HeahDude**
Fixes #7538.

On top of #8232, until this merge please just review b1e658e.

### CODE REVIEW

**Comment by HeahDude** [comment]
#8232 has been merged, rebased here.

File: service_container/3.3-di-changes.rst [hunk 1]
```diff
@@ -474,6 +476,69 @@ Many autoconfigured tags have an optional priority. If you need to specify a pri
 (or any other optional tag attribute), no problem! Just :ref:`manually configure your service <services-manually-wire-args>`
 and add the tag. Your tag will take precedence over the one added by auto-configuration.
 
+5) Auto-configure with _instanceof
+----------------------------------
+
+And the final big change is ``_instanceof``, it acts as a default definition
+template (see `service-33-default_definition`_), but only for service whose
```
Comment (xabbuh) [inline service_container/3.3-di-changes.rst]: services

File: service_container/3.3-di-changes.rst [hunk 2]
```diff
@@ -474,6 +476,69 @@ Many autoconfigured tags have an optional priority. If you need to specify a pri
 (or any other optional tag attribute), no problem! Just :ref:`manually configure your service <services-manually-wire-args>`
 and add the tag. Your tag will take precedence over the one added by auto-configuration.
 
+5) Auto-configure with _instanceof
+----------------------------------
+
+And the final big change is ``_instanceof``, it acts as a default definition
```
Comment (xabbuh) [inline service_container/3.3-di-changes.rst]: Let's start a new sentence with `it` to split this long one.

File: service_container/3.3-di-changes.rst [hunk 3]
```diff
@@ -474,6 +476,69 @@ Many autoconfigured tags have an optional priority. If you need to specify a pri
...
+template (see `service-33-default_definition`_), but only for service whose
+class matches a defined one.
+This can be very useful when many services shares some tag that cannot be
```
Comment (xabbuh) [inline service_container/3.3-di-changes.rst]: share

File: service_container/3.3-di-changes.rst [hunk 4]
```diff
@@ -474,6 +476,69 @@ Many autoconfigured tags have an optional priority. If you need to specify a pri
...
+inherited from abstract definition:
```
Comment (xabbuh) [inline service_container/3.3-di-changes.rst]: "[...] form abstract definitions" or "[...] from an abstract definition"

**Comment by HeahDude** [comment]
@xabbuh I've addressed your comments, thanks!

**Review by xabbuh — APPROVED** [review]

File: service_container/3.3-di-changes.rst [hunk 5]
```diff
@@ -474,6 +476,69 @@ Many autoconfigured tags have an optional priority. If you need to specify a pri
...
+        $definition
+            ->setAutowired(true)
+            ->setAutoconfigured(true)
+            ->setPublic(false)
+            ->setInstanceofConditionals(
+                'AppBundle\Domain\LoaderInterface' => $domainLoaderDefinition,
+            )
+        ;
+
+        $this->registerClasses($definition, 'AppBundle\\', '../../src/AppBundle/{Entity,Repository}');
```
Comment (xabbuh) [inline service_container/3.3-di-changes.rst]: we don't have this in the YAML or XML example

File: service_container/3.3-di-changes.rst [hunk 6]
```diff
@@ -474,6 +476,69 @@ Many autoconfigured tags have an optional priority. If you need to specify a pri
...
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <services>
+                <!-- ... -->
+
+                <instanceof class="AppBundle\Domain\LoaderInterface" public="true">
```
Comment (B-Galati) [inline service_container/3.3-di-changes.rst]: Looks like it's `id` instead of `class` based of my research in the codebase with 3.3

**Comment by xabbuh** [comment]
will be finished in #8496

**Pull request closed without merging**